plugins/com.google.cloud.tools.eclipse.appengine.libraries/src/com/google/cloud/tools/eclipse/appengine/libraries/Pom.java

```diff
@@ -135,12 +153,24 @@ void updateDependencies(Collection<Library> selectedLibraries,
     // m2e-core/org.eclipse.m2e.core.ui/src/org/eclipse/m2e/core/ui/internal/actions/AddDependencyAction.java
     // m2e-core/org.eclipse.m2e.core.ui/src/org/eclipse/m2e/core/ui/internal/editing/AddDependencyOperation.java
     
-    NodeList dependenciesList = document.getElementsByTagName("dependencies");
+    XPath xpath = xpathFactory.newXPath();
+    xpath.setNamespaceContext(new Maven4NamespaceContext());
+    
     Element dependencies;
-    if (dependenciesList.getLength() > 0) {
-      dependencies = (Element) dependenciesList.item(0);
-    } else {
-      dependencies = document.createElement("dependencies");
+    try {
+      NodeList dependenciesNodes = (NodeList) xpath.evaluate(
+          "./m:dependencies", // top-level elements only
+          document.getDocumentElement(),
+          XPathConstants.NODESET);
+      if (dependenciesNodes.getLength() > 0) {
+        dependencies = (Element) dependenciesNodes.item(0);
+      } else {
+        dependencies = document.createElementNS(
+            "http://maven.apache.org/POM/4.0.0", "dependencies");
+      }
+    } catch (XPathExpressionException ex) {
+      IStatus status = StatusUtil.error(Pom.class, ex.getMessage(), ex);
```

Comment:
This doesn't seem it will happen. `RuntimeException`?

Reply:
since we can return a status here, that's safer. But your'e right, it shouldn't happen.